doc/introduction/remote.rst

```diff
@@ -0,0 +1,511 @@
+
+and even saved as a new Blackbird script using the :func:`~.io.save` function:
+
+>>> from strawberryfields.io import save
+>>> save("test_compiled.xbb", prog)
+
+
+Tips and tricks
+---------------
+
+.. code-block:: python3
+
+    from strawberryfields.utils import operation
+
+We can define an operation to make it easier to apply the same unitary
+to both signal and idler modes.
+
+.. code-block:: python3
+
+    @operation(4)
+    def unitary(q):
+        ops.Interferometer(U) | q
+        ops.BSgate(0.543, 0.123) | (q[2], q[0])
+
+    prog = sf.Program(8)
+
+    with prog.context as q:
+        ops.S2gate(1.0) | (q[0], q[4])
+        ops.S2gate(1.0) | (q[1], q[5])
+        ops.S2gate(1.0) | (q[2], q[6])
+        ops.S2gate(1.0) | (q[3], q[7])
+
+        unitary() | q[:4]
+        unitary() | q[4:]
+
+
+Embedding bipartite graphs
+--------------------------
+
+We can embed bipartite graphs, with the restriction that the singular
+values form the set :math:`\{0, d\}` for some real value :math:`d`.
+
+The matrix :math:`B` represents the edges between the two sets of
+vertices in the graph, and :math:`A` is the full adjacency matrix
+:math:`A = \begin{bmatrix}0 & B\\ B^T & 0\end{bmatrix}`. Here, we will
+consider a complete bipartite graph, since we know that the singular
+values are of the form :math:`\{d, 0\}`.
+
+.. code-block:: python3
+
+    B = np.ones([4, 4])
+    A = np.block([[0*B, B], [B.T, 0*B]])
+
+    prog = sf.Program(8)
+
+    # the following mean photon number per mode
+    # quantity is set to ensure that the singular values
+    # are scaled such that all squeezers have value 1
+    m = 0.345274461385554870545
+
+    with prog.context as q:
+        ops.BipartiteGraphEmbed(A, mean_photon_per_mode=m) | q
+        ops.MeasureFock() | q
+
+
+>>> prog.compile("chip2").print()
+S2gate(1, 0) | (q[0], q[4])
+S2gate(0, 0) | (q[3], q[7])
+S2gate(0, 0) | (q[2], q[6])
+MZgate(3.598, 5.444) | (q[2], q[3])
+MZgate(3.598, 5.444) | (q[6], q[7])
+S2gate(0, 0) | (q[1], q[5])
+MZgate(0, 5.236) | (q[0], q[1])
+MZgate(4.886, 5.496) | (q[1], q[2])
+MZgate(0.7106, 4.492) | (q[2], q[3])
+Rgate(0.9284) | (q[3])
+MZgate(2.922, 3.142) | (q[0], q[1])
+MZgate(4.528, 3.734) | (q[1], q[2])
+Rgate(-2.51) | (q[2])
+MZgate(0, 5.236) | (q[4], q[5])
+MZgate(4.886, 5.496) | (q[5], q[6])
+MZgate(0.7106, 4.492) | (q[6], q[7])
+Rgate(0.9284) | (q[7])
+MZgate(2.922, 3.142) | (q[4], q[5])
+MZgate(4.528, 3.734) | (q[5], q[6])
+Rgate(-2.51) | (q[6])
+Rgate(-2.51) | (q[1])
+Rgate(-0.8273) | (q[4])
+Rgate(-0.8273) | (q[0])
+Rgate(-2.51) | (q[5])
+MeasureFock | (q[0], q[1], q[2], q[3], q[4], q[5], q[6], q[7])
+
+The squeezing values required to embed this bipartite graph are given by
+the following relation:
+
+>>> from thewalrus.quantum import find_scaling_adjacency_matrix
+>>> c = find_scaling_adjacency_matrix(A, 2*4*0.345274461385554870545)
+>>> set(np.arctanh(np.linalg.svd(c*A)[1]))
+{0.0, 1.0000000000000002}
```

Comment:
Might be better to put this earlier, where the comment that states why a specific m value was chosen is